lib/ansible/cli/galaxy.py

```diff
@@ -1466,6 +1487,11 @@ def execute_list_collection(self, artifacts_manager=None):
         for w in warnings:
             display.warning(w)
 
+        if output_format == 'json':
+            display.display(json.dumps(collections_in_paths))
+        elif output_format == 'yaml':
+            display.display(yaml.safe_dump(collections_in_paths))
+
```

Comment:
If no collection paths are found, this will output an empty `{}` before the help and error message. This should be after the `raise AnsibleOptionsError` below.

```
> env ANSIBLE_COLLECTIONS_PATH='' ansible-galaxy collection list --format json
{}
usage: ansible-galaxy [-h] [--version] [-v] TYPE ...

Perform various Role and Collection related operations.

positional arguments:
  TYPE
    collection   Manage an Ansible Galaxy collection.
    role         Manage an Ansible Galaxy role.

optional arguments:
  --version      show program's version number, config file location, configured module search path, module location, executable location and exit
  -h, --help     show this help message and exit
  -v, --verbose  verbose mode (-vvv for more, -vvvv to enable connection debugging)
ERROR! - None of the provided paths were usable. Please specify a valid path with --collections-path
```

Reply:
Why? Doesn't it go to stdout while help goes to stderr? With typical piping, it'd be alright and the tools down the chain would be able to consume and empty "there's nothing installed" `{}`...

Reply:
Awesome find. Fixed + added test.

Reply:
The empty JSON object was being included in stdout along with the usage text. IMO this _was_ a bug / undesirable behavior.

Reply:
I didn't know if it was intentional to output an empty dict. I don't think it will causes issues, just seems like it shouldn't be doing that.

Reply:
Demonstration:

```
$ ANSIBLE_COLLECTIONS_PATH='' ansible-galaxy collection list --format yaml 2> /dev/null
{}
usage: ansible-galaxy [-h] [--version] [-v] TYPE ...

Perform various Role and Collection related operations.

positional arguments:
  TYPE
    collection   Manage an Ansible Galaxy collection.
    role         Manage an Ansible Galaxy role.

optional arguments:
  --version      show program's version number, config file location, configured module search path, module location, executable location and exit
  -h, --help     show this help message and exit
  -v, --verbose  verbose mode (-vvv for more, -vvvv to enable connection debugging)
```

Reply:
Some options I can identify:

- Leave it as-is (patch w/ Sam's suggestion)
- We stop using `AnsibleOptionsError` here: https://github.com/ansible/ansible/blob/649df97d9174a285bcb08b881c45d2af4058d450/lib/ansible/cli/galaxy.py#L1491
- Make `AnsibleOptionsError` send to stderr
- We change the conditional that throws the error to be `if not path_found and output_format != 'human'`.

Reply:
The use of `AnsibleOptionsError` is to make in consistent with other sub commands. 

It looks like the help and `{}` will be it stdout. The error message is in `stderr`.

Reply:
So the fact that the help is in the `stdout` could cause issues. If the help went to `stderr`, that would be better in this case, but I don't think we should change that.

Reply:
Ok. Let's roll with it how it is now.